### DISCUSSION TITLE
IVFFlat vector, order by changes the number of results

### DISCUSSION THREAD

**Discussion by sebastian-weisshaar** [supabase/supabase#29091] [Questions]
DROP FUNCTION IF exists company_search5(vector,text[],company_source[]);

CREATE OR REPLACE FUNCTION company_search5 (
  query_embedding VECTOR(1536),
  countries text[],
  sources company_source[]
)
returns setof company
LANGUAGE plpgsql
SET statement_timeout TO '120s'  -- set custom timeout
AS $$
BEGIN
  set local ivfflat.probes = 23;
  RETURN QUERY
  SELECT 
    company.*
  FROM 
    public.company company
  INNER JOIN 
    public.company_enrichment enrichment on enrichment.company_id = company.id
  WHERE 
    company.country = ANY(countries) 
  AND
    company.source = ANY(sources)
  AND (enrichment.web_embedding <#> query_embedding) < -0.2
  ORDER BY (enrichment.web_embedding <#> query_embedding)
  LIMIT 1000;
END;
$$;

If I call this function from an authenticated users I get a different number of rows back compares to the Postgres role. The RLS on company and enrichment allows select for authenticated users. How can there still be a difference?

**Comment by GaryAustin1** [comment]
Have you tested with impersonation in the SQL editor?
It is quite common early on in dev using Supabase auth, especially with SSR, to not really have a user session in a client making a call even though signed in elsewhere in the app.
